File: src/main/java/io/curiousoft/ijudi/ordermanagement/repo/ProfileRepo.java
package io.curiousoft.ijudi.ordermanagement.repo;

import io.curiousoft.ijudi.ordermanagement.model.Profile;
import io.curiousoft.ijudi.ordermanagement.model.ProfileRoles;
import org.springframework.data.mongodb.repository.MongoRepository;

import java.util.List;

public interface ProfileRepo<U extends Profile> extends MongoRepository<U, String> {

    boolean existsByMobileNumber(String id);

    List<U> findByRole(ProfileRoles role);
}

File: izinga-commons/src/main/kotlin/commons/repo/ProfileRepo.kt
package io.curiousoft.izinga.commons.repo

import io.curiousoft.izinga.commons.model.Profile
import io.curiousoft.izinga.commons.model.ProfileRoles
import org.springframework.data.mongodb.repository.MongoRepository

interface ProfileRepo<U : Profile?> : MongoRepository<U, String?> {
    fun existsByMobileNumber(id: String?): Boolean
    fun findByRole(role: ProfileRoles?): List<U>?
}